fpop/run_fp.py

```diff
@@ -0,0 +1,197 @@
+from abc import ABC,abstractmethod
+from contextlib import contextmanager
+from dflow.python import (
+    OP,
+    OPIO,
+    OPIOSign,
+    Artifact,
+    TransientError,
+    FatalError,
+    BigParameter,
+    Parameter,
+)
+from dflow import (
+    Workflow,
+    Step,
+    upload_artifact,
+    download_artifact,
+    InputArtifact,
+    OutputArtifact,
+    ShellOPTemplate
+)
+import os, json, dpdata, shutil
+from pathlib import Path
+from typing import (
+    Any,
+    Tuple,
+    List,
+    Set,
+    Dict,
+    Optional,
+    Union,
+)
+import numpy as np
+import dargs
+from dargs import (
+    dargs,
+    Argument,
+    Variant,
+    ArgumentEncoder,
+)
+
+@contextmanager
+def set_directory(path: Path):
+    '''Sets the current working path within the context.
+    Parameters
```

Comment:
There should be a blank line be fore each section

Reply:
I have added the blank lines.